packages/common/src/compiler/translator.ts

```diff
@@ -140,6 +140,15 @@ const convertDbtMetricToLightdashMetric = (
             `Cannot determine model from dbt metric "${metric.name}" metric.model is "${metric.model}" but should be "ref('my_model')"`,
         );
     }
+    let sql = defaultSql(metric.name);
+    if (metric.sql) {
+        const isSingleColumnName = /^[a-zA-Z0-9_]+$/g.test(metric.sql);
+        if (isSingleColumnName) {
+            sql = defaultSql(metric.sql);
+        } else {
```

Comment:
This is going to fail in some instances where the `metric.sql` contains an ambiguous reference (when there are joined tables in the query). Still it's better than the current support and the error that appears will be very clear that it's a problem with ambiguous references.

Reply:
Yup, there will be some caveats when the user writes custom sql. I think there will always be caveats when users are using custom sql.